synapse/rest/client/login.py

```diff
@@ -342,6 +342,15 @@ async def _complete_login(
             user_id = canonical_uid
 
         device_id = login_submission.get("device_id")
+
+        # If device_id is present, check that device_id is not longer than a reasonable 512 characters
+        if device_id and len(device_id) > 512:
+            raise LoginError(
+                400,
+                "Device_id cannot be longer than 512 characters.",
```

Comment:
```suggestion
                "device_id cannot be longer than 512 characters.",
```

Our query parameters are case-sensitive, for better or worse :upside_down_face: 

Alternatively, if you want to keep the capital first letter:

```suggestion
                "Parameter 'device_id' cannot be longer than 512 characters.",
```